More fixes

belldandu opened the pull request:
users was broken and labeled as a dict it is now a list
g_participants now pushes to users and gets its info from _frame or frame.js
participant now pushes to users and gets its info from _frame or frame.js
show_fw now gets its info from _frame or frame.js
tb now gets its info from _frame or frame.js
show_tb now gets its info from _frame or frame.js
premium now gets its info from _frame or frame.js
also fixes have been added and applied to frame.js as well and they have all been tested :D
and are known to work :D 
Bansearchresult completed
blocklist fixed
blocked fixed 
unblocked fixed
premium updated
working on deleted error
fixed problems with recognition of whether the bot was a mod or admin or not 
fixed problems with ban list ban list back to old format and now works 
fixed an issue with spliting arguments that had ; in them in the banlist
fixed delmsg
added delUser
fixed clearUser
fixed flag
added flagUser
fixed ban
fixed unban
fixed problems with message system where everything was undefined >.>
other major fixes which were too numerous to put here 
lib is now at a stable version if you want you can update the version number
:D

belldandu committed:
More fixes

users was broken and labeled as a dict it is now a list
g_participants now pushes to users and gets its info from _frame or frame.js
participant now pushes to users and gets its info from _frame or frame.js
show_fw now gets its info from _frame or frame.js
tb now gets its info from _frame or frame.js
show_tb now gets its info from _frame or frame.js
premium now gets its info from _frame or frame.js
also fixes have been added and applied to frame.js as well and they have all been tested :D
and are known to work :D 
Bansearchresult completed
blocklist fixed
blocked fixed 
unblocked fixed
premium updated
working on deleted error

belldandu committed:
change the way self._bans 

was being added to 
it now does self._bans = {ban.username: ban}

belldandu committed:
Update frame.js

belldandu committed:
More fixes

blocklist fixed 
bansearchresult completed
unblocked fixed
blocked fixed
still working on deleted error

belldandu committed:
Update ch_room.js

bans changed again still having issues listing the bans though i dont think its grabbing the blocklist correctly so ill take another look

belldandu committed:
more fixes

fixed problems with recognition of whether the bot was a mod or admin or not 
fixed problems with ban list ban list back to old format and now works 
fixed an issue with spliting arguments that had ; in them in the banlist

belldandu committed:
more fixes 

fixed problems with recognition of whether the bot was a mod or admin or not 
fixed problems with ban list ban list back to old format and now works 
fixed an issue with spliting arguments that had ; in them in the banlist

belldandu commented:
final update tonight i fixed alot of stuff this patch should definetly be merged as both ch_room.js and frame.js need this update

MakuraYami commented:
Thanks a bunch, great to see someone is working with it :3 i don't have time to read it fully now, took a quick look and as i said before my local version has most of this done and more changes but its not done, and I'm currently too busy with different projects. As soon as I get some time I'll merge my latest code with your fixes

belldandu commented:
oh
i need one more small fix

belldandu committed:
Update ch_room.js

changed self._bansearch to this._bansearch

belldandu committed:
Update ch_room.js

fixing small stuff now xD

belldandu commented:
there we go i think that should just about do it xD

belldandu commented:
lol having a few probs with searchban but everything else is working xD

belldandu commented:
i might make a rooms thing but probably not now since i have no idea how to make the bot leave the room after its joined other than restarting the script

belldandu commented:
banning still doesn't work but you can check the ban list now and u can get the names of all the users currently in the chat as well as numerous other things so all in all i feel i've done well for a mediocre in javascript although i did learn python first so its not all that different the regular expressions are a pain but meh im getting used to them :P

belldandu committed:
Update frame.js

odd blank spot for ban search result now im just working on time

belldandu commented:
Last update for tonight I'm tired so yeah XD

belldandu committed:
another fix 

missed something last night
bansearch is hopefully fixed now

belldandu committed:
more updates

mkay so bansearch no longer a dict it just gets sent straight into the chat since the dict is always empty no matter what i try 
will most likely make the last few digits to ip replaced with xx to protect users
frame.js updated again as well

belldandu committed:
update again

update needed for ch_room.js added

belldandu commented on derplib/frame.js:
```diff
@@ -138,6 +138,44 @@ var frameTypesRoom = {
 	},
+
+	bansearchresult: function(_noidea, name, ip, unid, bansrc, timep1, timep2, timep3){
+		var time = new Date(Date.parse(String(String(timep1.split(' ').join('T'))+':'+String(timep2)+':'+String(timep3)+'.000')));
```
gg do not remove this

belldandu commented on derplib/frame.js:
```diff
@@ -138,6 +138,44 @@ var frameTypesRoom = {
 			time: parseFloat(time)};
 	},
+
+	bansearchresult: function(_noidea, name, ip, unid, bansrc, timep1, timep2, timep3){
```
the time has : in it twice so it tries to split it >.> sadly

belldandu commented on derplib/frame.js:
```diff
@@ -138,6 +138,44 @@ var frameTypesRoom = {
+			ip: ip,
+			unid: unid,
+			bansrc: bansrc,
+			time: String(time)};
```
yes it must be a string or its gonna be the date without it being in a string which is gonna cause error's so do not remove

belldandu committed:
Major Update

users is back to a dict now
it updates in i b participants and g_participants
ban and unban now works thanks to my fix to users
various other things are fixed
applying fix to frame.js as well again

belldandu commented on derplib/ch_room.js:
```diff
@@ -558,15 +574,15 @@ Room.prototype.clearall = function() {
 	if(this._isModerator) {
-		if(this._bans[user.name]){
-			this.write(['removeblock', this.bans[user.name].id, this.bans[user.name].ip]); 
+		if(this._bans[user].username){
```
username is not a typo do not change

belldandu commented on derplib/ch_room.js:
```diff
@@ -558,15 +574,15 @@ Room.prototype.clearall = function() {
-		if(this._bans[user.name]){
-			this.write(['removeblock', this.bans[user.name].id, this.bans[user.name].ip]); 
+		if(this._bans[user].username){
+			this.write(['removeblock', this._bans[user].id, this._bans[user].ip, this._bans[user].username]); 
```
username is not a typo do not change

belldandu committed:
Major Update

major fixes required for users to work correctly

belldandu committed:
Major Update (small fix)

:D small fix

belldandu commented:
lots of stuff fixed
this should definetly be merged at this point xD

belldandu committed:
last small fix

at this point this should be merged as ive fixed almoust everything so far and i dont plan to work on this again till next week

belldandu commented:
im done for a week this should be merged as soon as possible 
every thing is stable for now

belldandu committed:
small fix litterally last 

forgot a (

belldandu commented:
i had forgot a ( its fixed now

belldandu committed:
fixed bug of undefined in searchban

im now done for a week

belldandu commented:
> .> im tired

MakuraYami commented:
Ok, ill look at it tomorrow :D

On Apr 2, 2014 9:29 PM, "kamijoutouma" notifications@github.com wrote:

> im done for a week this should be merged as soon as possible
> every thing is stable for now
> 
> ## 
> 
> Reply to this email directly or view it on GitHub.

belldandu commented:
On a side note everything but delete works correctly all I have to do is make a function for it that let's me find the message id using the user name I also need to make msg.deleted the first half of that msg defined cause its not defined the first part ill try to tackle by making messages a object and see what I can do from there but that will most likely be when i have time (either Sunday and Monday or next week) but yeah everything else works :D

belldandu commented:
like self._messages[user] = [messages] and then var messages;
var message = { user: message.user,
time: message.time,
id: message.id,
?: ?,
?: ?
}
return messages.push(message);
Something along those lines also I did this from phone so it looks crappy but you get the general idea xD

belldandu commented:
Lol I forgot body: message.body xD

belldandu commented:
Examples but the concept is easy to understand xD

MakuraYami commented on derplib/frame.js:
```diff
@@ -71,7 +71,7 @@ var frameTypesRoom = {
 	},
 	
-	"delete": function(msgid) {
+	delete: function(msgid) {
```
delete is a js keyword thats why it had quotes. This is bad mkay :(

belldandu replied:
mkay ill fix it xD

belldandu committed:
Update frame.js

belldandu commented:
fixed

belldandu committed:
fixes to mods structure

had error's on removing and adding mods xD

belldandu committed:
fix for msg.deleted

lol cant beleive i didnt notice that

belldandu committed:
multi fixes needed for ch_room.js

>.>

belldandu committed:
more fixes :D

major fixes to delete and flag as well as message system

belldandu commented:
there now the lib should be stable :D 
deleted was fixed

belldandu committed:
lol 

reversed the order in which it checked the messages for delete and flag

belldandu committed:
again

little fixes

belldandu committed:
fixes

lol

belldandu committed:
fixes

more fixes small

belldandu committed:
fixes

more fixes small

belldandu commented:
that last fix is final fix for this version of derplib no more fixes needed till u add more stuff

belldandu commented:
also id is set to undefined so that an actual value can be placed there cause false is a boolean and can only be overwritten by true

belldandu committed:
added a room manager

you can now disconnect from rooms and there is a rooms list which the name of the room is added to 
please note this is not complete you cant disconnect from a room if you are not using the command in that room

belldandu committed:
i was way off

anyway imma let you do the room manager thingy xD for now the disconnect method is there

belldandu commented:
gg im done for now
